data/saas/dataset/segment_dataset.yml

```diff
@@ -7,8 +7,6 @@ dataset:
         fields:
           - name: segment_id
             data_categories: [user.unique_id]
-            fidesops_meta:
-              primary_key: True
```

Comment:
Did we test that erasure was called after removing `primary_key` from this field? I don't think the `data_protection_request` will work if none of the fields are flagged as `primary_key`

Reply:
For some reason, it needed to be moved to the `traits` collection from what I could tell, so did it here:
https://github.com/ethyca/fides/pull/5004/files#diff-eb9b94a980a356ef500f8905ac01f617398e36f2e0123fe92d7b441b4a6894ceR98-R99